### PR TITLE
Adds support for resizing replicas if using kube v1 APIs

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes: https://github.com/projectatomic/atomicapp/issues/147

 This patch adds provision for lookup of resource identity based on the API
 version under processing. Presently support for v1, v1beta3 and v1beta1 APIs
 are included.

### CODE REVIEW

**Comment by vpavlin** [comment]
LGTM

**Comment by goern** [comment]
@navidshaikh could you pls extend https://github.com/projectatomic/atomicapp/blob/master/Providers.asciidoc#features-1 with two three sentences... thx!

File: Providers.asciidoc [hunk 1]
```diff
@@ -51,6 +51,9 @@ There are no configuration values for the Docker provider.
 The Kubernetes Provider supports namespaces, this will deploy an
 application into a given namespace. All components of the applications
 will be created in that namespace.
+The `namespace` defined in Kubernetes artificats, if any, must match with the namespace value in configuration.
+
+Note: In Kubernetes, the default value of `namespace` field (if not defined in Kubernetes manifests explicitly) is `default` and one may leave the following configuration value to its default value.
```
Comment (navidshaikh) [inline Providers.asciidoc]: @goern : Does this make sense? or we need to re-phrase it?

Comment (goern) [inline Providers.asciidoc]: ja :) but, I thought ... maybe... you could.. describe the resizing behavior in Providers.asciidoc

**Comment by navidshaikh** [comment]
Since latest kube version we have in vagrant box only supports `v1` and we are moving all the examples to `v1`, I think this patch needs to be updated to parse only `v1` APIs and not the others like `v1beta3` or `v1beta1`, etc.

@goern : thoughts ?

File: tests/cached_nulecules/guestbook-go/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM projectatomic/atomicapp:0.1.1
```
Comment (goern) [inline tests/cached_nulecules/guestbook-go/Dockerfile]: could you make this 0.1.3 at least :)

File: tests/cached_nulecules/guestbook-go/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM projectatomic/atomicapp:0.1.3
```
Comment (navidshaikh) [inline tests/cached_nulecules/guestbook-go/Dockerfile]: @goern 
>could you make this 0.1.3 at least :)
done :-)

File: Providers.asciidoc [hunk 4]
```diff
@@ -61,6 +64,19 @@ will be created in that namespace.
...
+`**atomicapp run**`
+
+This command deploys the app in Kubernetes cluster in specified namespace. For the given namespace, the deploy process creates objects (pods, replicas, services) in order as enlisted in Nulecule Kubernetes artifacts.
+
+`**atomicapp stop**`
+
+This command undeploys the app in Kubernetes cluster in specified namespace. For the given namespace, the undeploy process consist of 
+
+ 1. scaling down the replicas to size 0
+ 2. deleting all objects (pods, replicas, services)
```
Comment (navidshaikh) [inline Providers.asciidoc]: @goern 
> ja :) but, I thought ... maybe... you could.. describe the resizing behavior in Providers.asciidoc

check this.

**Pull request merged**